test/functional/feature_fee_estimation.py

```diff
@@ -391,6 +400,85 @@ def test_acceptstalefeeestimates_option(self):
         assert_equal(self.nodes[0].estimatesmartfee(1)["feerate"], fee_rate)
 
 
+    def sanity_check_cpfp_estimates(self, utxos):
+        """The BlockPolicyEstimator currently does not take CPFP into account. This test
+        sanity checks its behaviour when receiving transactions that were confirmed because
+        of their child's feerate.
+        """
+        # The broadcaster and block producer
+        node = self.nodes[0]
+        miner = self.nodes[1]
+        # In sat/vb
+        low_feerate = Decimal(2)
+        med_feerate = Decimal(15)
+        high_feerate = Decimal(20)
+
+        # If a transaction got mined and one of its descendants has a higher ancestor
+        # score, it does not get taken into account by the fee estimator.
+        tx = send_tx(self.wallet, node, None, low_feerate)
+        u = {"txid": tx["txid"], "vout": 0, "value": Decimal(tx["tx"].vout[0].nValue) / COIN}
+        send_tx(self.wallet, node, u, high_feerate)
+        self.sync_mempools(wait=0.1, nodes=[node, miner])
+        self.generate(miner, 1)
+        assert node.estimaterawfee(1)["short"]["fail"]["totalconfirmed"] == 0
+
+        # If it has descendants which have a lower ancestor score, it does though.
+        tx = send_tx(self.wallet, node, None, high_feerate)
+        u = {"txid": tx["txid"], "vout": 0, "value": Decimal(tx["tx"].vout[0].nValue) / COIN}
+        send_tx(self.wallet, node, u, low_feerate)
+        self.sync_mempools(wait=0.1, nodes=[node, miner])
+        self.generate(miner, 1)
+        assert node.estimaterawfee(1)["short"]["fail"]["totalconfirmed"] == 1
+
+        # Same if it's equal.
+        tx = send_tx(self.wallet, node, None, high_feerate)
+        u = {"txid": tx["txid"], "vout": 0, "value": Decimal(tx["tx"].vout[0].nValue) / COIN}
+        send_tx(self.wallet, node, u, high_feerate)
+        self.sync_mempools(wait=0.1, nodes=[node, miner])
+        self.generate(miner, 1)
+        # Decay of 0.962, truncated to 2 decimals in the RPC result
+        assert node.estimaterawfee(1)["short"]["fail"]["totalconfirmed"] == Decimal("1.96")
```

Comment:
this seems very brittle, can you add a formula to recalculate this if something changes, or use the formula directly here?